src/platform/vita/vita_keyboard.c

```diff
@@ -0,0 +1,173 @@
+    return -1;
+
+    // Convert UTF8 to UTF16
+    memset(ime_title_utf16, 0, sizeof(ime_title_utf16));
+    memset(ime_initial_text_utf16, 0, sizeof(ime_initial_text_utf16));
+    utf8_to_utf16((uint8_t *)title, ime_title_utf16);
+    utf8_to_utf16((uint8_t *)initial_text, ime_initial_text_utf16);
+
+    //clear previous results
+    memset(ime_input_text_utf16, 0, sizeof(ime_input_text_utf16));
+    memset(ime_input_text_utf8, 0, sizeof(ime_input_text_utf8));
+
+    SceImeDialogParam param;
+    sceImeDialogParamInit(&param);
+
+    param.supportedLanguages = 0x0001FFFF;
+    param.languagesForced = SCE_TRUE;
+    param.type = type;
+    param.option = option;
+    if (option == SCE_IME_OPTION_MULTILINE)
+        param.dialogMode = SCE_IME_DIALOG_DIALOG_MODE_WITH_CANCEL;
+    param.title = ime_title_utf16;
+    param.maxTextLength = max_text_length;
+    param.initialText = ime_initial_text_utf16;
+    param.inputTextBuffer = ime_input_text_utf16;
+
+    int res = sceImeDialogInit(&param);
+    if (res >= 0) {
+        ime_dialog_running = 1;
+        ime_dialog_option = option;
+    }
+
+    return res;
+}
+
+static uint8_t *get_ime_dialog_input_text_utf8(void)
+{
+    return ime_input_text_utf8;
+}
+
+static int update_ime_dialog(void) {
+    if (!ime_dialog_running)
+    return IME_DIALOG_RESULT_NONE;
+
+    SceCommonDialogStatus status = sceImeDialogGetStatus();
+    if (status == IME_DIALOG_RESULT_FINISHED) {
+        SceImeDialogResult result;
+        memset(&result, 0, sizeof(SceImeDialogResult));
+        sceImeDialogGetResult(&result);
+
+        if ((ime_dialog_option == SCE_IME_OPTION_MULTILINE && result.button == SCE_IME_DIALOG_BUTTON_CLOSE) ||
+        (ime_dialog_option != SCE_IME_OPTION_MULTILINE && (result.button == SCE_IME_DIALOG_BUTTON_ENTER || result.button == SCE_IME_DIALOG_BUTTON_CLOSE))) {
+            // Convert UTF16 to UTF8
+            utf16_to_utf8(ime_input_text_utf16, ime_input_text_utf8);
+        } else {
+            status = IME_DIALOG_RESULT_CANCELED;
+        }
+
+        sceImeDialogTerm();
+
+        ime_dialog_running = 0;
+    }
+
+    return status;
+}
+
+char *vita_keyboard_get(char *title, const char *initial_text, int maxLen, int multiline)
+{
+    char *name = NULL;
+
+    if (ime_init_apputils == 0) {
+        sceAppUtilInit(&(SceAppUtilInitParam){}, &(SceAppUtilBootParam){});
+        sceCommonDialogSetConfigParam(&(SceCommonDialogConfigParam){});
+        ime_init_apputils = 1;
+    }
+    if (multiline)
+    init_ime_dialog(title, initial_text, maxLen, SCE_IME_TYPE_BASIC_LATIN, SCE_IME_OPTION_MULTILINE);
+    else
+    init_ime_dialog(title, initial_text, maxLen, SCE_IME_TYPE_BASIC_LATIN, 0);
+
+    bool done;
+    do {
+        vita2d_start_drawing();
+        vita2d_clear_screen();
+
+        done = true;
+
+        int ime_result = update_ime_dialog();
+        if (ime_result == IME_DIALOG_RESULT_FINISHED) {
+            name = (char *)get_ime_dialog_input_text_utf8();
```

Comment:
Doesn't it work to return `name` here instead of using the `done` boolean?

Reply:
No, I remember testing this, and the vita2d_commands have to be executed, otherwise I got gfx flicker in Rocks'n'Diamonds at least.